tests/regression/test_vertex_based_limiter.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+from firedrake import *
+from firedrake.slope_limiter.vertex_based_limiter import VertexBased
+import numpy as np
+
+
+def test_constant_field(mesh_size=30):
```

Comment:
Do these tests also work on periodic intervals?  If so, you can use pytest parametrisation to write the combinations for you:

```python
@pytest.fixture(params=["periodic-interval", "periodic-square-tri", "periodic-square-quad"])
def mesh(request):
    if request.param == "periodic-interval":
        return PeriodicUnitIntervalMesh(30)
    elif request.param == "periodic-square-tri":
        return PeriodicUnitSquareMesh(30, 30)
    elif request.param == "periodic-square-quad":
        return PeriodicUnitSquareMesh(30, 30, quadrilateral=True)
```

And then you just write:

```python
def test_constant_field(mesh):
    ...
```

And pytest will inject the right mesh into the function call for you.